gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java

```diff
@@ -134,6 +129,58 @@ private ClusterStats getPrestoClusterStats(ProxyBackendConfiguration backend) {
         conn.disconnect();
       }
     }
+    return null;
+  }
+
+  private ClusterStats getPrestoClusterStats(ProxyBackendConfiguration backend) {
+    ClusterStats clusterStats = new ClusterStats();
+    clusterStats.setClusterId(backend.getName());
+
+    // Fetch Cluster level Stats.
+    String target = backend.getProxyTo() + UI_API_STATS_PATH;
+    String response = queryCluster(target);
+    if (response == null) {
```

Comment:
check isNullOrEmpty instead? stringbuilder output will not be null

Reply:
done, the null check was because the queryCluster method returns null in-case of an error but null + empty would be better.